js/main.js

```diff
@@ -17,8 +17,63 @@ function onVersionChange(form, animation) {
         });
     });
 }
+
 function loadVersions() {
     let form = $('[data-form-name$="_event"]');
     onVersionChange(form);
     form.find('#id_c_versions').change(function () { onVersionChange(form, 'slow') });
 }
+
+function hideExtraVersions(form) {
+    $.each(version_prefixes, function(version, prefix) {
+        if (prefix == 'kr_' || prefix == 'tw_'){
+	    let input = form.find('#id_c_versions [value="' + version + '"]');
+	    let field = input.closest('label');
+            field.hide('slow');
+            input.val('');
+    }
+    });
+}
+
+function onLocationChange(form){
+        let checkbox_h=form.find('#id_c_locations [value="hits"]');
+        let checkbox_d=form.find('#id_c_locations [value="daily"]');
+        let checkbox_b=form.find('#id_c_locations [value="bside"]');
```

Comment:
Use explicit variable names: `checkbox_h` -> `checkbox_hits`.

Spaces around `=`:
```javascript
let checkbox_hits = form.find(...);
```

Reply:
Maybe you can simplify the whole logic of this function by using a foreach that goes through the variables and their related fields:

```javascript
$.each({
    hits: ['unlock'],
    bside: ['b_side_master', 'b_side_start', 'b_side_end'],
    ...
}, function(checkbox_name, related_fields) {
    let checked = form.find('#id_c_locations [value="' + checkbox_name + '"]').prop('checked');
    $.each(related_fields, function(i, field_name) {
        let input = form.find('#id_' + field_name);
        let field = input.closest('form-group');
        if (checked) {
            field.show('fast');
        } else {
            input.val('');
            field.hide('fast');
        }
    });
});
```

I didn't actually check that the code runs, you'll need to check that 😅

Also you could make it easier to update in the future by providing the values in python:

```python
# models.py:

class Song(...):
    ...
    LOCATIONS = [
        ('hits', {
            'translation': _('Hits'),
            'related_fields': ['unlock'],
        }),
        ...
    ]
    LOCATIONS_CHOICES = [(_name, _info['translation']) for _name, _info in LOCATIONS.items()]
    LOCATIONS_RELATED_FIELDS = [(_name, _info['related_fields']) for _name, _info in LOCATIONS.items()]
    c_locations = models.TextField(_('Locations'), blank=True, null=True)

# magicollections.py:

    def _modification_extra_context(self, context):
        ...
        context['js_variables']['locations_related_fields'] = jsv(models.Song.LOCATIONS_RELATED_FIELDS)
```

Then you can just do:

```javascript
$.each(locations_related_fields, function(checkbox_name, related_fields) {
    ...
```

Reply:
The code needed to be adjusted, but I got a more loop-y version working! ✅